jbang bouncinglogo@jbangdev fails to copy images

similar to #534 

`❯ jbang --verbose --fresh bouncinglogo@jbangdev
[jbang] Reading aliases from /Users/max/.jbang/implicit-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/quarkusio/jbang-catalog/master/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/quarkusio/jbang-catalog/blob/master/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/cache/urls/c79114b08f217b977a92049e80c21d920728b32cbc0944906eb8c616b32bda71/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/zemiak/javakarel/master/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/zemiak/javakarel/blob/master/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/cache/urls/4b822aa749af0c178d2ebf105a515ac0c216bf976939b2d879aaf9e8bc2adc0e/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/jruby/jbang-catalog/master/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/jruby/jbang-catalog/blob/master/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/cache/urls/23afa01792d8488c13620869b43c9b882d6a771e60564100ad69c40a03288f93/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/maxandersen/jbang-catalog/master/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/maxandersen/jbang-catalog/blob/master/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/cache/urls/12cee842a085318cb370d234493f93def3225bf51f011de2d691f643e5f1d75a/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/jacoco/jbang-catalog/main/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/jacoco/jbang-catalog/blob/main/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/cache/urls/ab08c62cd98c3ec7a71e132fe0bb7e536db7ee98387adf5c00bb546ad54f71f5/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/jacoco/jbang-catalog/HEAD/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/jacoco/jbang-catalog/blob/HEAD/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/cache/urls/f95e1762ebee995668e90d0709f1e2e530695f2d1b06d45befc5bf1c29a5d81f/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/linux-china/jbang-catalog/HEAD/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/linux-china/jbang-catalog/blob/HEAD/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/cache/urls/639c6b25c4ccbe4c3e35692c026de0db6e3dfd4d86e2eaa04ce4dd8fb996c816/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/jbangdev/jbang-catalog/master/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/jbangdev/jbang-catalog/blob/master/jbang-catalog.json
[jbang] Reading aliases from /Users/max/.jbang/cache/urls/22276b0e71b9dc0e466e9158ad103a5d9175aaadaed34eeb3bb6c8b25b80ddde/jbang-catalog.json
[jbang] Retrieved file from cache https://raw.githubusercontent.com/jbangdev/jbang-catalog/master/jbang-catalog.json = /Users/max/.jbang/cache/urls/22276b0e71b9dc0e466e9158ad103a5d9175aaadaed34eeb3bb6c8b25b80ddde/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/jbangdev/jbang-catalog/blob/master/jbang-catalog.json
[jbang] Reading aliases from /Users/max/code/personal/jbang/jbang-catalog.json
[jbang] Retrieved file from cache https://raw.githubusercontent.com/jbangdev/jbang-catalog/master/jbang-catalog.json = /Users/max/.jbang/cache/urls/22276b0e71b9dc0e466e9158ad103a5d9175aaadaed34eeb3bb6c8b25b80ddde/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/jbangdev/jbang-catalog/blob/master/jbang-catalog.json
[jbang] Retrieved file from cache https://raw.githubusercontent.com/jbangdev/jbang-catalog/master/jbang-catalog.json = /Users/max/.jbang/cache/urls/22276b0e71b9dc0e466e9158ad103a5d9175aaadaed34eeb3bb6c8b25b80ddde/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/jbangdev/jbang-catalog/blob/master/jbang-catalog.json
[jbang] Retrieved file from cache https://raw.githubusercontent.com/jbangdev/jbang-catalog/master/jbang-catalog.json = /Users/max/.jbang/cache/urls/22276b0e71b9dc0e466e9158ad103a5d9175aaadaed34eeb3bb6c8b25b80ddde/jbang-catalog.json
[jbang] Obtained catalog from https://github.com/jbangdev/jbang-catalog/blob/master/jbang-catalog.json
[jbang] Downloaded file https://raw.githubusercontent.com/jbangdev/jbang-catalog/master/bouncinglogo.java
[jbang] System Java version detected as 11
[jbang] System Java version matches requested version 11
[jbang] Resolving dependencies...
[jbang]     Resolving org.openjfx:javafx-controls:11.0.2:mac...Done
[jbang]     Resolving org.openjfx:javafx-graphics:11.0.2:mac...Done
[jbang] Dependencies resolved
[jbang] Copying https:/raw.githubusercontent.com/jbangdev/jbang-catalog/master/images/jbang-icon.png to /Users/max/.jbang/cache/jars/bouncinglogo.java.4d3567eda5cfc4d6217a11b57f140bdb647e2aacc91fe9fade734ce44eec18b4/images/jbang-icon.png
[jbang] Deleting /Users/max/.jbang/cache/jars/bouncinglogo.java.4d3567eda5cfc4d6217a11b57f140bdb647e2aacc91fe9fade734ce44eec18b4
[jbang] [ERROR] Could not copy https:/raw.githubusercontent.com/jbangdev/jbang-catalog/master/images/jbang-icon.png to /Users/max/.jbang/cache/jars/bouncinglogo.java.4d3567eda5cfc4d6217a11b57f140bdb647e2aacc91fe9fade734ce44eec18b4/images/jbang-icon.png
dev.jbang.ExitException: Could not copy https:/raw.githubusercontent.com/jbangdev/jbang-catalog/master/images/jbang-icon.png to /Users/max/.jbang/cache/jars/bouncinglogo.java.4d3567eda5cfc4d6217a11b57f140bdb647e2aacc91fe9fade734ce44eec18b4/images/jbang-icon.png
	at dev.jbang.FileRef.copy(FileRef.java:68)
	at dev.jbang.cli.BaseBuildCommand.buildJar(BaseBuildCommand.java:172)
	at dev.jbang.cli.BaseBuildCommand.build(BaseBuildCommand.java:129)
	at dev.jbang.cli.Run.prepareArtifacts(Run.java:70)
	at dev.jbang.cli.Run.doCall(Run.java:59)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:80)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:12)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1933)
	at picocli.CommandLine.access$1200(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
	at picocli.CommandLine.execute(CommandLine.java:2058)
	at dev.jbang.Main.main(Main.java:14)
Caused by: java.nio.file.NoSuchFileException: https:/raw.githubusercontent.com/jbangdev/jbang-catalog/master/images/jbang-icon.png
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixCopyFile.copy(UnixCopyFile.java:548)
	at java.base/sun.nio.fs.UnixFileSystemProvider.copy(UnixFileSystemProvider.java:258)
	at java.base/java.nio.file.Files.copy(Files.java:1294)
	at dev.jbang.FileRef.copy(FileRef.java:66)
	... 14 more
[jbang] If you believe this a bug in jbang open issue at https://github.com/jbangdev/jbang/issues
```


...notice its treating the url as a file not url.


another weirdness...why is it contacting that many catalogs?

--- a/itests/jbang-catalog.json
+++ b/itests/jbang-catalog.json
@@ -1,4 +1,10 @@
 {
+  "catalogs": {
+    "test": {
+      "catalog-ref": "https://raw.githubusercontent.com/jbangdev/jbang/master/itests/jbang-catalog.json",
+      "description": "JBang test scripts"
+    }
+  },
   "aliases": {
     "hello": {
       "script-ref": "hello.jsh"

--- a/itests/run.feature
+++ b/itests/run.feature
@@ -31,6 +31,12 @@ Scenario: java run multiple files using alias
   When command('jbang resource')
   Then match out contains "hello properties"
 
+Scenario: java run multiple files using remote alias
+  When command('jbang catalog add test https://raw.githubusercontent.com/jbangdev/jbang/master/itests/jbang-catalog.json')
+  Then command('jbang trust add https://raw.githubusercontent.com')
+  Then command('jbang resource@test')
+  Then match out contains "hello properties"
+
 Scenario: java run with agent
   When command('jbang --verbose --javaagent=JULAgent.java=options JULTest.java World')
   Then match err contains "info World"

--- a/src/main/java/dev/jbang/FileRef.java
+++ b/src/main/java/dev/jbang/FileRef.java
@@ -34,7 +34,7 @@ public class FileRef {
 			throw new IllegalStateException("Only relative paths allowed in //FILES. Found absolute path: " + p);
 		}
 
-		return Paths.get(source.getScriptResource().getOriginalResource()).resolveSibling(p);
+		return source.getOriginalFile().toPath().resolveSibling(p);
 	}
 
 	protected Path to(Path parent) {

--- a/src/main/java/dev/jbang/Script.java
+++ b/src/main/java/dev/jbang/Script.java
@@ -40,14 +40,10 @@ public class Script {
 	private static final String REPOS_ANNOT_PREFIX = "@GrabResolver(";
 	private static final Pattern REPOS_ANNOT_PAIRS = Pattern.compile("(?<key>\\w+)\\s*=\\s*\"(?<value>.*?)\"");
 	private static final Pattern REPOS_ANNOT_SINGLE = Pattern.compile("@GrabResolver\\(\\s*\"(?<value>.*)\"\\s*\\)");
+
 	private final ScriptResource scriptResource;
 
-	/**
-	 * The original reference, it might or might not be same as used as backingFile.
-	 * TODO: should probably have a "originalRef" to capture GAV+main ref and a
-	 * "originalFile" which could be null.
-	 */
-	private String originalFile;
+	private String originalRef;
 
 	private ModularClassPath classpath;
 	private String script;
@@ -266,7 +262,7 @@ public class Script {
 							Collections.emptyList(), offline, !Util.isQuiet());
 				} else {
 					if (getBackingFile() == null) {
-						classpath = new ModularClassPath(Arrays.asList(new ArtifactInfo(null, new File(originalFile))));
+						classpath = new ModularClassPath(Arrays.asList(new ArtifactInfo(null, getOriginalFile())));
 					} else {
 						classpath = new ModularClassPath(Arrays.asList(new ArtifactInfo(null, getBackingFile())));
 					}
@@ -458,12 +454,38 @@ public class Script {
 		return getBackingFile().getName().endsWith(".jsh");
 	}
 
-	public void setOriginal(String probe) {
-		this.originalFile = probe;
+	public void setOriginal(String ref) {
+		this.originalRef = ref;
 	}
 
-	public String getOriginalFile() {
-		return originalFile;
+	/**
+	 * The original script reference. Might ba a URL or an alias.
+	 */
+	public String getOriginalRef() {
+		return originalRef;
+	}
+
+	/**
+	 * The resource that `originalRef` resolves to. Wil be a URL or file path.
+	 */
+	public String getOriginalResource() {
+		return scriptResource.getOriginalResource();
+	}
+
+	/**
+	 * The resource that `originalRef` resolves to. Will be a URL or file path.
+	 */
+	public File getOriginalFile() {
+		return new File(getOriginalResource());
+	}
+
+	/**
+	 * The actual local file that `originalRef` refers to. This might be the sane as
+	 * `originalRef` if that pointed to a file on the local file system, in all
+	 * other cases it will refer to a downloaded copy in Jbang's cache.
+	 */
+	public File getBackingFile() {
+		return scriptResource.getFile();
 	}
 
 	public void createJarFile(File path, File output) throws IOException {
@@ -528,10 +550,6 @@ public class Script {
 		return getBackingFile() != null && getBackingFile().toString().endsWith(".jar");
 	}
 
-	public File getBackingFile() {
-		return scriptResource.getFile();
-	}
-
 	public ModularClassPath getClassPath() {
 		return classpath;
 	}
@@ -553,7 +571,7 @@ public class Script {
 		if (isURL(fileReference)) {
 			return new URLRef(source, ref, dest);
 		}
-		if (isURL(source.originalFile)) {
+		if (isURL(source.getOriginalResource())) {
 			return new URLRef(source, ref, dest);
 		} else {
 			return new FileRef(source, ref, dest);

--- a/src/main/java/dev/jbang/URLRef.java
+++ b/src/main/java/dev/jbang/URLRef.java
@@ -24,7 +24,7 @@ public class URLRef extends FileRef {
 		}
 
 		try {
-			return new URI(source.getScriptResource().getOriginalResource()).resolve(p);
+			return new URI(source.getOriginalResource()).resolve(p);
 		} catch (URISyntaxException e) {
 			throw new IllegalStateException("Could not resolve URI", e);
 		}

--- a/src/main/java/dev/jbang/cli/Edit.java
+++ b/src/main/java/dev/jbang/cli/Edit.java
@@ -83,9 +83,9 @@ public class Edit extends BaseScriptCommand {
 			out.println("echo " + project.getAbsolutePath()); // quit(project.getAbsolutePath());
 		} else {
 			try (final WatchService watchService = FileSystems.getDefault().newWatchService()) {
-				File orginalFile = new File(script.getOriginalFile());
+				File orginalFile = script.getOriginalFile();
 				if (!orginalFile.exists()) {
-					throw new ExitException(2, "Cannot live edit " + script.getOriginalFile());
+					throw new ExitException(2, "Cannot live edit " + script.getOriginalRef());
 				}
 				Path watched = orginalFile.getAbsoluteFile().getParentFile().toPath();
 				watched.register(watchService,
@@ -125,7 +125,7 @@ public class Edit extends BaseScriptCommand {
 	/** Create Project to use for editing **/
 	File createProjectForEdit(Script script, boolean reload) throws IOException {
 
-		File originalFile = new File(script.getOriginalFile());
+		File originalFile = script.getOriginalFile();
 
 		List<String> collectDependencies = script.collectDependencies();
 		String cp = script.resolveClassPath(offline);

--- a/src/main/java/dev/jbang/cli/Info.java
+++ b/src/main/java/dev/jbang/cli/Info.java
@@ -35,7 +35,7 @@ abstract class BaseInfoCommand extends BaseScriptCommand {
 			List<String> collectDependencies = script.collectDependencies();
 			String cp = script.resolveClassPath(offline);
 
-			originalResource = script.getOriginalFile();
+			originalResource = script.getOriginalResource();
 			backingResource = script.getBackingFile().toString();
 
 			if (cp.isEmpty()) {

--- a/src/main/java/dev/jbang/cli/Run.java
+++ b/src/main/java/dev/jbang/cli/Run.java
@@ -194,7 +194,7 @@ public class Run extends BaseBuildCommand {
 						}
 						if (jar == null) {
 							throw new ExitException(EXIT_INTERNAL_ERROR,
-									"No jar found for agent " + agent.getOriginalFile());
+									"No jar found for agent " + agent.getOriginalRef());
 						}
 						fullArgs.add("-javaagent:" + jar
 								+ (agent.getJavaAgentOption() != null ? "=" + agent.getJavaAgentOption() : ""));

